source/nodejs/adaptivecards/src/serialization.ts

```diff
@@ -197,7 +197,9 @@ export abstract class BaseSerializationContext {
             propertyValue === undefined ||
             propertyValue === defaultValue
         ) {
-            delete target[propertyName];
+            if (!GlobalSettings.enableFullJsonRoundTrip) {
```

Comment:
For these methods, I mimicked the behavior of `serializeValue`. Is there a reason why we wouldn't want to move this check to be one level up?

The only difference is that we would be reassigning `target[propertyName]`